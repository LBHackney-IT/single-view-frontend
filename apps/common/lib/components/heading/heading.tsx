import React, { forwardRef } from "react";

import cn from "classnames";

import type * as Polymorphic from "@radix-ui/react-polymorphic";

import "./styles.scss";

export interface HeadingProps {
  variant?: "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
}

export type HeadingComponent = Polymorphic.ForwardRefComponent<"h1", HeadingProps>;

export const Heading: HeadingComponent = forwardRef(function Heading(
  { as: HeadingComp, variant = "h1", className, ...props },
  ref,
) {
  const Comp = HeadingComp || variant;
  return (
    <Comp ref={ref} className={cn(`lbh-heading-${variant}`, className)} {...props} />
  );
});
