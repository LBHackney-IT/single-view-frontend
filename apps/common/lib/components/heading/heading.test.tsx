import React from "react";

import { render, testA11y } from "@hackney/mtfh-test-utils";

import { Heading } from "./heading";

test("it renders the heading", async () => {
  const { container } = render(<Heading>Heading</Heading>);
  expect(container).toMatchSnapshot();
  await testA11y(container);
});

test("it renders the heading as h2 styles", async () => {
  const { container } = render(<Heading as="h2">Heading</Heading>);
  expect(container).toMatchSnapshot();
});

test("it renders the heading as h2 with h3 styles", async () => {
  const { container } = render(
    <Heading as="h2" variant="h3">
      Heading
    </Heading>,
  );
  expect(container).toMatchSnapshot();
});
