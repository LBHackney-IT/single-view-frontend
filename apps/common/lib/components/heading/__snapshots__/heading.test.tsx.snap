// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders the heading 1`] = `
<div>
  <h1
    class="lbh-heading-h1"
  >
    Heading
  </h1>
</div>
`;

exports[`it renders the heading as h2 styles 1`] = `
<div>
  <h2
    class="lbh-heading-h1"
  >
    Heading
  </h2>
</div>
`;

exports[`it renders the heading as h2 with h3 styles 1`] = `
<div>
  <h2
    class="lbh-heading-h3"
  >
    Heading
  </h2>
</div>
`;
